lua/core/menu/devices.lua

```diff
@@ -110,26 +130,55 @@ m.enc = function(n,delta)
   end
 end
 
+local function redraw_keyboard_layout_select()
+  local len = tabutil.count(m.options["keyboard layout"])
+  screen.clear()
+  for i=1,6 do
+    if (i > 2 - m.pos + 1) and (i < m.len - m.pos + 3 + 1) then
+      local name = m.options["keyboard layout"][i+m.pos-2 - 1]
+      local line = string.upper(name)
+      screen.level(i==3 and 15 or 4)
+      screen.move(20,10*i)
+      screen.text(line)
+    end
+  end
+  screen.update()
+end
+
 m.redraw = function()
+
+  if (m.section == "keyboard layout" and m.mode == "select") then
+    redraw_keyboard_layout_select()
```

Comment:
i didn't want to modify current display/scroll beahviour of other submenus so made a dedicated redraw function